sktime/performance_metrics/forecasting/_functions.py

```diff
@@ -46,6 +48,20 @@
 EPS = np.finfo(np.float64).eps
 
 
+def _get_kwarg(kwarg_, **kwargs):
+    kwarg_ = kwargs.pop(kwarg_, None)
+    if kwarg_ is None:
+        calling_function = inspect.stack()[1].function
```

Comment:
I remember reading up on how to get the name of the calling function a while ago and came across a few warnings using something like this (see e.g. here https://stackoverflow.com/questions/2654113/how-to-get-the-callers-method-name-in-the-called-method). I don't fully understand all the consequences, but a simpler option may be to just pass the name (or some context object) into the function, may not be as elegant but potentially more reliable/efficient (see e.g. `check_window_length` in `utils.validation.forecasting`). 

```python
def _get_kwarg(**kwargs, kwarg, name):
    ...
    msg = f"{name} requires ..."
    ...
```

Reply:
ah, I also would recommend not to do this - function behaviour should be fully defined by arguments passed to it.
Lookup of calling environment etc just invites hard-to-detect errors to happen...

Reply:
@mloning and @fkiraly, I"m with you on this. Reading more about it, it definately seems like we want to avoid it. 

However, manually passing the name seems like a bit of a pain (albeit a minor one), so I considered one of the other solutions from stackoverflow. What are your thoughts about defining a decorator? 

The following works to pass metric name to the metrics that need it. 

```python
from functools import wraps
def expose_metric_name(func):
    @wraps(func)
    def tmp(*args, metric_name=func.__name__, **kwargs):
        return func(*args, metric_name=metric_name, **kwargs)
    return tmp


def _get_kwarg(kwarg, metric_name="Metric", **kwargs):
    kwarg_ = kwargs.pop(kwarg, None)
    if kwarg_ is None:
        msg = "".join(
            [
                f"{metric_name} requires `{kwarg}`.",
                f"Pass `{kwarg}` as a keyword argument when calling the metric.",
            ]
        )
        raise ValueError(msg)
    return kwarg_


@expose_metric_name
def mean_absolute_scaled_error(
    y_true, y_pred, sp=1, horizon_weight=None, multioutput="uniform_average", **kwargs
):
    metric_name = kwargs.pop("metric_name", None)
    y_train = _get_kwarg("y_train", metric_name=metric_name, **kwargs)
    ... # rest of function
```

Reply:
That seems like a nice solution and I'm also not happy about having to pass the name into the function, but I don't think the code is very readable anymore. Compare below:
```python
def mean_absolute_scaled_error(
    y_true, y_pred, sp=1, horizon_weight=None, multioutput="uniform_average", **kwargs
):
    y_train = _get_kwarg("y_train", name="mean_absolute_scaled_error", **kwargs)
    ... # rest of function
```
The name is also unlikely to change in the future I guess. Alternatively, we could simply raise an error without reference to the function name, that may be the simplest solution.

Reply:
I have no problem adding the names -- I've pushed an update that includes those; i'd rather the error message was more informative with the name. 

I was more interested in testing out a way to do that more automatically. Would have been satisfying to be able to have one, but I agree that using the decorator makes that a bit hard to read.

Reply:
Okay great!